src/core/check.rs

```diff
@@ -367,36 +368,34 @@ pub fn infer(
       exp_typ.whnf(defs);
       match exp_typ.head {
         DAGPtr::Slf(_) => {
-          let new_link = match exp_typ.clone().head {
+          let new_link = match exp_typ.head {
             DAGPtr::Slf(link) => link,
             _ => panic!("Error in the clone implementation"),
           };
           let Slf { var: new_var, bod: new_bod, .. } =
             unsafe { &mut *new_link.as_ptr() };
           if new_var.parents.is_some() {
-            let exp_clone = exp.clone();
             replace_child(
               DAGPtr::Var(NonNull::new(new_var).unwrap()),
-              exp_clone.head,
+              exp.head,
             );
           }
           replace_child(DAGPtr::Slf(new_link), *new_bod);
           let new_bod = DAG::new(*new_bod);
           Ok((exp_ctx, new_bod))
         }
-        _ => Err(CheckError::GenericError(format!(
+        _ => Err(CheckError::GenericError(
           "Tried to case on something that isn't a datatype."
-        ))),
+        .to_owned())),
       }
     }
     DAGPtr::All(link) => {
       let All { var, dom, img, .. } = unsafe { &mut *link.as_ptr() };
       let mut typ = DAG::from_term(&Term::Typ(None));
       let dom = DAG::new(*dom);
-      let _ = check(defs, pre.clone(), Uses::None, &dom, &mut typ)?;
-      let dom_clone = dom.clone();
+      std::mem::drop(check(defs, pre.clone(), Uses::None, &dom, &mut typ)?);
       (*var).dep = pre.len() as u64;
-      pre.push((var.nam.to_string(), dom_clone));
+      pre.push((var.nam.to_string(), dom));
```

Comment:
In this specific case, `dom` was not used after the creation of `dom_clone`, so the clone was unnecessary, This means that `dom`can be used instead of `dom_clone` since `dom_clone` is only used once.